brownfield_django/main/models.py

```diff
@@ -136,11 +136,19 @@ def role(self):
             return "student"
 
     def get_absolute_url(self):
+        url = '/'
         if self.is_teacher():
-            return '/ccnmtl/home/%s/' % (self.id)
+            url = '/ccnmtl/home/%s/' % (self.id)
         if self.is_admin():
```

Comment:
yeah. a user can never be a student and teacher at the same time so this shouldn't cause any problems, but I will change it to an elif for clarity.